File: Other Demo/CALayer/CALayer4.2/ViewController.m
//
//  ViewController.m
//  CALayer4.2
//
//  Created by yubinqiang on 15/9/17.
//  Copyright © 2015年 Kingsoft. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *snowmanImageView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.snowmanImageView.layer.borderWidth = 3;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)changeColor:(UIButton *)sender {
}

@end

File: Other Demo/CALayer/CALayer7.5/ViewController.swift
//
//  ViewController.swift
//  CALayer7.5
//
//  Created by Milo on 2017/10/27.
//  Copyright © 2017年 Kingsoft. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
    }

    @IBAction func handleTouchUpInside(_ sender: UIButton) {
        
        print("action OUTSIDE :\(String(describing: self.view.layer.action(forKey: "backgroundColor")))")
        
        UIView.animate(withDuration: 0.25) {
            print("action INSIDE :\(String(describing: self.view.layer.action(forKey: "backgroundColor")))")
        }
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

